source/loader/layers/sanitizer/asan/asan_interceptor.cpp

```diff
@@ -887,6 +914,28 @@ bool ProgramInfo::isKernelInstrumented(ur_kernel_handle_t Kernel) const {
     return InstrumentedKernels.find(Name) != InstrumentedKernels.end();
 }
 
+/**
+ * Relase the actual allocation without other checks
+ */
+ur_result_t
+AsanInterceptor::releaseAllocationNoCheck(ur_context_handle_t Context,
+                                          std::shared_ptr<AllocInfo> AI,
+                                          bool IsFromQuarantined) {
+    if (IsFromQuarantined) {
+        getContext()->logger.info("Quarantine Free: {}",
+                                  (void *)AI->AllocBegin);
+    } else {
+        getContext()->logger.info("Free: {}", (void *)AI->AllocBegin);
+    }
```

Comment:
I feel not comfortable that "IsFromQuarantined" is just for different logger, it'd be better to move logger before calling "releaseAllocationNoCheck".

Reply:
That is just a helper function to reduce duplicated codes. If we split that logic to caller, then we would have to repeat the same code in different places, which is quite again the purpose of the function.

Reply:
I mean you can just move "getContext()->logger" before calling this function.